synapse/storage/databases/state/store.py

```diff
@@ -404,6 +406,111 @@ def _insert_into_cache(
                 fetched_keys=non_member_types,
             )
 
+    async def store_state_deltas_for_batched(
```

Comment:
This function both stores state groups to the DB and updates the event contexts that are passed in, is that an appropriate 
approach?

Reply:
The mutated fields are read elsewhere in the code and the updated values are important, so it seems like it might make sense to run this chunk of code again after the transaction. The reason I do it within the transaction is that otherwise it would be messy/difficult to update the all the tables in one transaction